server/src/main/java/com/netflix/conductor/server/ConductorConfig.java

```diff
@@ -95,24 +95,37 @@ public String getAvailabilityZone() {
 
 	@Override
 	public int getIntProperty(String key, int defaultValue) {
-		String val = System.getProperty(key);
-		int value  = defaultValue;
+		String val = getProperty(key, Integer.toString(defaultValue));
 		try{
-			value = Integer.parseInt(val);
-		}catch(Exception e){}
-		return value;
+			defaultValue = Integer.parseInt(val);
+		}catch(NumberFormatException e){}
+		return defaultValue;
 	}
 
 	@Override
 	public String getProperty(String key, String defaultValue) {
-		return Optional.ofNullable(System.getProperty(key)).orElse(defaultValue);
+
+		String val = null;
+		try{
+			val = System.getenv(key.replace('.','_'));
```

Comment:
Some docker orchestrators will replace `.` with `_`.  Nomad for example.   Maybe we could check both?  For now the assumption is that config values look like `component.subsection.key` and environment variables take the form `component_subsection_key`.